turf/line_intersect/_line_intersect.py

```diff
@@ -0,0 +1,173 @@
+
+    :param line_1: line_1 coordinates of segments
+    :param line_2: line_2 coordinates of segments
+    :returns: list of line segments that possibly intersect with each other
+    """
+    possible_intersects = []
+    rtree_index = index.Index()
+
+    for i, seg in enumerate(line_1):
+
+        bbox = envelope(seg)
+        rtree_index.insert(i, bbox["bbox"])
+
+    for j, seg in enumerate(line_2):
+
+        bbox = envelope(seg)
+        seg_intersection_idx = deque(rtree_index.intersection(bbox["bbox"]))
+
+        while seg_intersection_idx:
+
+            seg_idx = seg_intersection_idx.pop()
+
+            possible_intersects.append([*line_1[seg_idx], *line_2[j]])
+
+    return possible_intersects
+
+
+def calculate_intersect(
+    seg_1_start: Sequence,
+    seg_1_end: Sequence,
+    seg_2_start: Sequence,
+    seg_2_end: Sequence,
+) -> Union[None, Point]:
+    """
+    Calculates the intersection point of two segments otherwise None
+
+    :param seg_1_start: coordinates of segment 1 start
+    :param seg_1_end: coordinates of segment 1 end
+    :param seg_2_start: coordinates of segment 2 start
+    :param seg_2_end: coordinates of segment 2 end
+    :returns: {Point} if intersection, otherwise None
+    """
+    x1 = seg_1_start[0]
+    y1 = seg_1_start[1]
+    x2 = seg_1_end[0]
+    y2 = seg_1_end[1]
+    x3 = seg_2_start[0]
+    y3 = seg_2_start[1]
+    x4 = seg_2_end[0]
+    y4 = seg_2_end[1]
+
+    pnt = None
+
+    denom = ((y4 - y3) * (x2 - x1)) - ((x4 - x3) * (y2 - y1))
+    numeA = ((x4 - x3) * (y1 - y3)) - ((y4 - y3) * (x1 - x3))
+    numeB = ((x2 - x1) * (y1 - y3)) - ((y2 - y1) * (x1 - x3))
+
+    if denom == 0:
+        return pnt
+
+    uA = numeA / denom
+    uB = numeB / denom
+
+    if uA >= 0 and uA <= 1 and uB >= 0 and uB <= 1:
+        x = round(x1 + (uA * (x2 - x1)), 6)
+        y = round(y1 + (uA * (y2 - y1)), 6)
+        pnt = point([x, y])
+
+    return pnt
+
+
+def get_line_segments(line: LinePolyFeature) -> Sequence:
+    """
+    Gets segments from a line feature
+
+    :param line: any LineString or Polygon
+    :return: sequence of segmetns
+    """
+
+    segments = []
+
+    try:
+        line_geometry = line.get("geometry").get("type")
+    except AttributeError:
+        try:
+            line_geometry = line["type"]
+        except KeyError:
+            raise InvalidInput(
+                error_code_messages["InvalidGeometry"](["Polygon", "MultiPolygon"])
+            )
+
+    if line_geometry in ["MultiPolygon", "Polygon"]:
+        line = polygon_to_line(line)
+        line_geometry = line["geometry"]["type"]
+
+    line_coords = get_coords_from_features(line, ("LineString", "MultiLineString"))
+
+    if line_geometry in ["LineString"]:
+        line_coords = [line_coords]
+
+    for line_coord in line_coords:
+        for idx in range(len(line_coord) - 1):
+            segments.append([line_coord[idx], line_coord[idx + 1]])
+
+        zip(line_coord, line_coord[1:])
```

Comment:
this line is not doing anything here. I would replace everyhting with:

```python
segments = [list(zip(coords, coords[1:])) for coords in line_coords]
```

Reply:
this is interesting because, on my local branch, I don't see this code line. no idea, how this appeared in the pr.

Reply:
also your suggestion fails later in the code.